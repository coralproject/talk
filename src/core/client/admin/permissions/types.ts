import { GQLUSER_ROLE_RL } from "coral-framework/schema";

export type PermissionMap<AbilityType extends string, ContextType> = {
  [abilityType in AbilityType]: {
    [role in GQLUSER_ROLE_RL]?: (context?: ContextType) => boolean;
  };
};
