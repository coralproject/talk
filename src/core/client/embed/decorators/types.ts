import pym from "pym.js";

export type CleanupCallback = () => void;
export type Decorator = (pym: pym.Parent) => CleanupCallback | void;
