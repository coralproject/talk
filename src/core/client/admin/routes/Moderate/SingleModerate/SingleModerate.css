$single-moderate-background: var(--v2-colors-grey-100);
$single-moderate-bar: var(--v2-colors-blue-500);
$single-moderate-bar-text: var(--v2-colors-pure-white);

.background {
  position: fixed;
  top: 0;
  left: 0;
  z-index: -1;
  width: 100%;
  height: 100%;
  background-color: $single-moderate-background;
}

.main {
  margin: calc(2 * var(--mini-unit)) 0 calc(4 * var(--mini-unit)) 0;
  display: flex;
  justify-content: center;
}

.subBar {
  height: calc(4 * var(--mini-unit));
  background-color: $single-moderate-bar;
  margin-top: -1px;
}

.subBarBegin {
  position: absolute;
  left: 0;
  font-family: var(--v2-font-family-primary);
  text-decoration: none;
  font-weight: var(--v2-font-weight-primary-semi-bold);
  font-size: var(--font-size-2);
  text-transform: uppercase;
  color: $single-moderate-bar-text;
}

.subBarTitle {
  font-family: var(--v2-font-family-primary);
  font-weight: var(--v2-font-weight-primary-semi-bold);
  font-size: var(--font-size-3);
  text-transform: uppercase;
  color: $single-moderate-bar-text;
}
