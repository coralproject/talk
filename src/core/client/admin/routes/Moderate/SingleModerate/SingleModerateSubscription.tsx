import { graphql, requestSubscription } from "react-relay";
import { Environment } from "relay-runtime";

import {
  createSubscription,
  SubscriptionVariables,
} from "coral-framework/lib/relay";

import { SingleModerateSubscription } from "coral-admin/__generated__/SingleModerateSubscription.graphql";

const SingleModerateSubscription = createSubscription(
  "subscribeToSingleModerate",
  (
    environment: Environment,
    variables: SubscriptionVariables<SingleModerateSubscription>
  ) =>
    requestSubscription(environment, {
      subscription: graphql`
        subscription SingleModerateSubscription($commentID: ID!) {
          commentStatusUpdated(id: $commentID) {
            comment {
              id
              status
              ...MarkersContainer_comment @relay(mask: false)
              ...ModeratedByContainer_comment @relay(mask: false)
            }
          }
        }
      `,
      variables,
      updater: store => {
        const commentID = store
          .getRootField("commentStatusUpdated")!
          .getLinkedRecord("comment")!
          .getValue("id")!;
        const commentInStore = store.get(commentID);
        if (commentInStore) {
          // Mark that the status of the comment was live updated.
          commentInStore.setValue(true, "statusLiveUpdated");
        }
      },
    })
);

export default SingleModerateSubscription;
