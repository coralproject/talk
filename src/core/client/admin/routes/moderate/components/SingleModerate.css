.background {
  position: fixed;
  top: 0;
  left: 0;
  z-index: -1;
  width: 100%;
  height: 100%;
  background-color: var(--palette-background-light);
}

.main {
  margin: calc(2 * var(--spacing-unit)) 0 calc(4 * var(--spacing-unit)) 0;
  display: flex;
  justify-content: center;
}

.subBar {
  height: calc(3 * var(--spacing-unit));
  background-color: var(--palette-primary-dark);
  margin-top: -1px;
}

.subBarBegin {
  position: absolute;
  left: 0;

  font-size: calc(12rem / var(--rem-base));
  font-weight: var(--font-weight-medium);
  font-family: var(--font-family-sans-serif);
  line-height: calc(14em / 12);
  letter-spacing: calc(0.2em / 12);

  color: var(--palette-common-white);
  text-transform: uppercase;
  text-decoration: none;
}

.subBarTitle {
  font-size: calc(14rem / var(--rem-base));
  font-weight: var(--font-weight-medium);
  font-family: var(--font-family-sans-serif);
  line-height: calc(16em / 14);
  letter-spacing: calc(0.2em / 14);

  color: var(--palette-common-white);
  text-transform: uppercase;
}
