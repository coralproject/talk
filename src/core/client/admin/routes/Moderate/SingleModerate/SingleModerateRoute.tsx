import { noop } from "lodash";
import React, { FunctionComponent, useEffect } from "react";
import { graphql } from "react-relay";

import { useSubscription } from "coral-framework/lib/relay";
import { withRouteConfig } from "coral-framework/lib/router";

import { SingleModerateRouteQueryResponse } from "coral-admin/__generated__/SingleModerateRouteQuery.graphql";

import NotFound from "../../NotFound";
import { LoadingQueue, Queue } from "../Queue";
import SingleModerate from "./SingleModerate";
import SingleModerateSubscription from "./SingleModerateSubscription";

type Props = SingleModerateRouteQueryResponse;

const danglingLogic = () => false;

const SingleModerateRoute: FunctionComponent<Props> = props => {
  const subscribeToSingleModerate = useSubscription(SingleModerateSubscription);
  useEffect(() => {
    if (!props.comment) {
      return;
    }
    const disposable = subscribeToSingleModerate({
      commentID: props.comment.id,
    });
    return () => {
      disposable.dispose();
    };
  }, [props.comment, subscribeToSingleModerate]);

  if (!props.comment) {
    return <NotFound />;
  }
  return (
    <SingleModerate>
      <Queue
        comments={[props.comment]}
        settings={props.settings}
        viewer={props.viewer!}
        onLoadMore={noop}
        hasLoadMore={false}
        disableLoadMore={false}
        danglingLogic={danglingLogic}
      />
    </SingleModerate>
  );
};

const enhanced = withRouteConfig<Props, SingleModerateRouteQueryResponse>({
  query: graphql`
    query SingleModerateRouteQuery($commentID: ID!) {
      comment(id: $commentID) {
        id
        ...ModerateCardContainer_comment
      }
      settings {
        ...ModerateCardContainer_settings
      }
      viewer {
        ...ModerateCardContainer_viewer
      }
    }
  `,
  cacheConfig: { force: true },
  render: function SingleModerateRouteRender({ Component, data }) {
    if (Component && data) {
      return <Component {...data} />;
    }
    return <LoadingQueue />;
  },
})(SingleModerateRoute);

export default enhanced;
