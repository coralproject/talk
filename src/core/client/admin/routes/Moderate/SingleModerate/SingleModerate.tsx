import { Localized } from "@fluent/react/compat";
import { Link } from "found";
import React, { FunctionComponent } from "react";

import MainLayout from "coral-admin/components/MainLayout";
import { SubBar } from "coral-ui/components/v2";

import styles from "./SingleModerate.css";

interface Props {
  children?: React.ReactNode;
}

const Moderate: FunctionComponent<Props> = ({ children }) => (
  <div data-testid="single-moderate-container">
    <SubBar className={styles.subBar} gutterBegin gutterEnd>
      <Localized id="moderate-single-goToModerationQueues">
        <Link className={styles.subBarBegin} to="/admin/moderate/">
          Go to moderation queues
        </Link>
      </Localized>
      <Localized id="moderate-single-singleCommentView">
        <div className={styles.subBarTitle}>Single Comment View</div>
      </Localized>
    </SubBar>
    <div className={styles.background} />
    <MainLayout>
      <main className={styles.main}>{children}</main>
    </MainLayout>
  </div>
);

export default Moderate;
