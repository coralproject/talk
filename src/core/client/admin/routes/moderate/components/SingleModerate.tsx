import { Localized } from "fluent-react/compat";
import { Link } from "found";
import React, { StatelessComponent } from "react";

import MainLayout from "talk-admin/components/MainLayout";
import { SubBar } from "talk-ui/components";

import styles from "./SingleModerate.css";

interface Props {
  children?: React.ReactNode;
}

const Moderate: StatelessComponent<Props> = ({ children }) => (
  <div data-test="single-moderate-container">
    <SubBar className={styles.subBar} gutterBegin gutterEnd>
      <Localized id="moderate-single-goToModerationQueues">
        <Link className={styles.subBarBegin} to="/admin/moderate/">
          Go to moderation queues
        </Link>
      </Localized>
      <Localized id="moderate-single-singleCommentView">
        <div className={styles.subBarTitle}>Single Comment View</div>
      </Localized>
    </SubBar>
    <div className={styles.background} />
    <MainLayout>
      <main className={styles.main}>{children}</main>
    </MainLayout>
  </div>
);

export default Moderate;
