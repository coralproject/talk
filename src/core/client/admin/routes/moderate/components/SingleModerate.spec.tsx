import { shallow } from "enzyme";
import React from "react";

import { PropTypesOf } from "talk-framework/types";

import SingleModerate from "./SingleModerate";

it("renders correctly", () => {
  const props: PropTypesOf<typeof SingleModerate> = {
    children: "singe comment queue",
  };
  const wrapper = shallow(<SingleModerate {...props} />);
  expect(wrapper).toMatchSnapshot();
});
