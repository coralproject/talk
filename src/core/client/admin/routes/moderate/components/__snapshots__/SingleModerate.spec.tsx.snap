// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  data-test="single-moderate-container"
>
  <withPropsOnChange(SubBar)
    className="SingleModerate-subBar"
    gutterBegin={true}
    gutterEnd={true}
  >
    <Localized
      id="moderate-single-goToModerationQueues"
    >
      <Link
        className="SingleModerate-subBarBegin"
        to="/admin/moderate/"
      >
        Go to moderation queues
      </Link>
    </Localized>
    <Localized
      id="moderate-single-singleCommentView"
    >
      <div
        className="SingleModerate-subBarTitle"
      >
        Single Comment View
      </div>
    </Localized>
  </withPropsOnChange(SubBar)>
  <div
    className="SingleModerate-background"
  />
  <MainLayout>
    <main
      className="SingleModerate-main"
    >
      singe comment queue
    </main>
  </MainLayout>
</div>
`;
